### ISSUE TITLE
Admin Accessibility: "Confirm Delete" button color contrast

### ISSUE THREAD

**Issue by aviannamiller** [omeka/omeka-s#2263]
In the Items list in the admin dashboard, there are two different button text:background color contrasts happening depending on whether you opt to delete a selection of items or delete all items. When tested with TPGi's Colour Contrast Analyser, the button to "Confirm Delete" all items has inadequate contrast for regular text with a ratio of 3.1:1. The text is #a7a7a7, and the background is #a91919. On the other hand, the button to "Confirm Delete" just a selection of items uses #fff text over the #a91919 background and has an adequate ratio of 7.4:1.

<img width="1212" alt="Screenshot of Colour Contrast Analyser window that lists 3.1:1 ratio overlaying an Omeka S admin dashboard with a 'Delete Items' panel open next to browser development tools that reveal content's text and background hex codes" src="https://github.com/user-attachments/assets/8eaa58ff-7fbd-44d1-aa16-1459b5c6faca" />

- Ratio should be corrected to at least 4.5:1
- Best solution would probably be to copy over style settings from the "Confirm Delete" button used for deleting a selection of items

**Comment by zerocrates** [comment]
Looks like a styling or cascade error that's making the text gray, fairly certain it's supposed to be white or near-white there.

**Comment by zerocrates** [comment]
Ah I see, it's gray because it's disabled (until you click the checkbox)

**Comment by aviannamiller** [comment]
Got it. Maybe something as simple as the background color switching instead of the text for better legibility (ex. dark grey to red)?

**Comment by zerocrates** [comment]
The basic problem is that it's a generic style change for all disabled elements that makes the text gray, and it doesn't account for the dark background used here.

I assume, yeah, the fix will look like a specific style just for this button when disabled, that does the background only, or changes both the background and the text, or something along those lines.

**Comment by kimisgold** [comment]
I've provided disabled button styles for the confirm delete all button. There should also now be screen reader alerts for toggling the "Are you sure?" checkbox, telling the user that it's activated or deactivated the "Confirm delete" button. It's ready to pull and test.

**Comment by aviannamiller** [comment]
I believe an `aria-disabled` attribute needs to be employed here in order to indicate for keyboard users that, while unchecked, there is a button that will need to be activated in order to submit the decision to delete all.

Testing using VoiceOver on Chrome, Safari, and Firefox. I think the readout creates a bit of redundancy and confusion. Maybe just adding the word "button" (as in: Confirm Delete **button** active/inactive) would better connect the checkbox and "Confirm Delete" button into the two step process that they are. However, this may not be necessary depending on the workflow after implementing `aria-disabled`, would have to revisit after that point.

Also, when unchecked, the "Confirm Delete" button is too low of a contrast (2.1:1). Though technically acceptable, I would recommend aiming for better legibility in order to better indicate that there is a second step that will need to be taken, in this case pressing the confirmation button. A greater contrast in greys should do the trick.

**Comment by zerocrates** [comment]
How are you proposing to use aria-disabled here? Just added to the button I don't believe it's going to do anything different at all; "real" `disabled` will expose that state to screenreaders just the same.

Or are you proposing removing the `disabled` attribute, adding aria-disabled, then doing the actual disabling with JS?

**Comment by aviannamiller** [comment]
The updated contrast looks good, and the amended alerts certainly lend more clarity to the process!

There may be a larger discussion in all of this for considering moving away from disabled buttons and replacing them with, as one example of an alternative, operable buttons that produce error messages when clicked that require confirmation by the user. Alternatively, tooltips could potentially be added to include instruction to the button, but I could see how this may remove some of the intentionality behind having the two step confirmation process.

In this case, however, I do not believe `disabled` reveals the button to screenreader users as it is unfocusable (tested with VoiceOver on Chrome, Firefox, and Safari). `disabled` should remain to inform the style of disabled buttons, but an `aria-disabled` attribute would also need to be introduced to bring that focus back. Can that be done here? @kimisgold has pointed out the possible overhaul required in introducing `aria-disabled`, depending on how it is done, so I am curious (especially thinking a bit more globally about how disabled buttons are used across S) what would be the best approach in terms of clarifying the process to screenreader users while understanding what is actually plausible in the code here.
